packages/utils/src/Registry/Registry.ts

```diff
@@ -1,13 +1,16 @@
+import { Middleware, compose, AnyAction } from 'redux';
+
 import ReducerRegistry from '../ReducerRegistry';
-export let registry;
 
-function init(initialState = {}, middleware = [], composeEnhancersDefault) {
+export let registry: ReducerRegistry<any>;
+
+function init(initialState: Record<string, unknown> = {}, middleware: Middleware[] = [], composeEnhancersDefault?: typeof compose) {
```

Comment:
Can you make the `initialState` argument generic? You can then use the generic type for all the state arguments.

I could look something like: 
```TS
function init<T extends Record<string, any>>(initialState: T = {} as T, ...) {
    if (!registry) {
    registry = new ReducerRegistry<T>(initialState, [...middleware], composeEnhancersDefault);
  }
}
...
  registry.register({
    routerData: (state: T, { type, payload }: AnyAction) => {
```